packages/contracts/test/router/v6/nftx/offers.test.ts

```diff
@@ -0,0 +1,353 @@
+                {
+                  orderIndex: i * 2,
+                  itemIndex: 0,
+                },
+              ],
+            })),
+          ],
+        ]),
+        value: 0,
+      },
+      // 2. Fill offers with the received NFTs
+      ...offers
+        .filter((_) => _.order)
+        .map((offer, i) => ({
+          module: nftxModule.address,
+          data: nftxModule.interface.encodeFunctionData("sell", [
+            [offer.order?.params],
+            {
+              fillTo: carol.address,
+              refundTo: carol.address,
+              revertIfIncomplete
+              // revertIfIncomplete: false,
+            },
+            [
+              ...fees[i].map((amount) => ({
+                recipient: emilio.address,
+                amount,
+              })),
+            ],
+          ]),
+          value: 0,
+        })),
+    ];
+
+    // Checks
+
+    // If the `revertIfIncomplete` option is enabled and we have any
+    // orders that are not fillable, the whole transaction should be
+    // reverted
+    if (
+      partial &&
+      revertIfIncomplete &&
+      offers.some(({ isCancelled }) => isCancelled)
+    ) {
+      await expect(
+        router.connect(carol).execute(executions, {
+          value: executions
+            .map(({ value }) => value)
+            .reduce((a, b) => bn(a).add(b), bn(0)),
+        })
+      ).to.be.revertedWith(
+        "reverted with custom error 'UnsuccessfulExecution()'"
+      );
+
+      return;
+    }
+
+    // Fetch pre-state
+    const balancesBefore = await getBalances(
+      Sdk.Common.Addresses.Weth[chainId]
+    );
+
+    // Execute
+
+    const tx = await router.connect(carol).execute(executions, {
+      value: executions
+        .map(({ value }) => value)
+        .reduce((a, b) => bn(a).add(b), bn(0)),
+    });
+
+    const txReceipt = await ethers.provider.getTransactionReceipt(tx.hash);
+
+    for (const log of txReceipt.logs) {
+      try {
+        const parsed = nftxModule.interface.parseLog(log);
+        if (parsed.name === "LogStr") {
+          const revertReason = ethers.utils.defaultAbiCoder.decode(
+            ["string"],
+            "0x" +
+              ethers.utils.defaultAbiCoder
+                .decode(["bytes"], `0x${log.data}`)[0]
+                .substring(10)
+          );
+          console.error("revertReason", revertReason);
+        }
+      } catch {
+        //
+      }
+    }
+
+    // Fetch post-state
+
+    const balancesAfter = await getBalances(Sdk.Common.Addresses.Weth[chainId]);
+
+    // Checks
+
+    // Carol got the payment
+    const carolAfter = balancesAfter.carol.sub(balancesBefore.carol);
+    const orderSum = offers
+      .map((offer, i) => (offer.isCancelled ? bn(0) : bn(offer.price)))
+      .reduce((a, b) => bn(a).add(b), bn(0));
+
+    const orderAmount = parseFloat(formatEther(orderSum)) 
+    const maxDiff = orderAmount * (orderAmount < 0.2 ? 0.8 : 0.3);
```

Comment:
Is there a way to enforce exact amounts? The tests should ideally be deterministic and also not depend on the inputs (which in case it does).

Reply:
there has some issue here. the fee has not been counted in! now change to slippage difference check.